### ISSUE TITLE
Sharing static files with default index.html in subpath

### ISSUE THREAD

**Issue by git-no** [httpswift/swifter#198]
Often websites url calling directories, the webserver does use the standard index.html automatically.

This is implemented for the root directory with commit [c92f8d1b8d6728f3fa8f4007cc9989d6cdc7fb8c](https://github.com/httpswift/swifter/commit/c92f8d1b8d6728f3fa8f4007cc9989d6cdc7fb8c) but it does not work for subdirectories.

I suggest this code change to achieve index.html is used for subdirectories when no file is called.

````
let url = URL(string: r.path)!
if let file = try? (directoryPath + String.pathSeparator + fileRelativePath.value + (url.hasDirectoryPath ? String.pathSeparator + "index.html" : "") ).openForReading() {
```
in exchange for 
```
if let file = try? (directoryPath + String.pathSeparator + fileRelativePath.value ).openForReading() {
```
in Swifter source file Files.swift

It worked in my tests. 
Attention: the url function .hasDirectoryPath is available since iOS 9.

**Comment by hangox** [comment]
It work!!! thanks！

**Comment by valdirunars** [comment]
Probably best to add mime type support as well, currently I think swifter only supports serving html and text so all <scripts /> and <link /> tags won't do what you think

**Comment by dev-journey** [comment]
Please, could you post some example code to access index.html file on a root subfolder, because I can access the index.html file that's directly located at root level:

    let server = HttpServer()
    let documentsUrl = FileManager.default.urls(for: .documentDirectory, in: .userDomainMask)[0]
    let indexFile = "index.html"
    let htmlTmp = documentsUrl.appendingPathComponent("root")
    server["/:path"] = shareFilesFromDirectory(htmlTmp.path, defaults: [indexFile])
    try! server.start(9990)
    let myRequest = URLRequest(url: URL(string: "http://127.0.0.1:9990/")!)
    webview.load(myRequest)

But can't access if index.html file is on a, for example, root/base/index.html.

**Comment by valdirunars** [comment]
yes but can you `<link>` to a stylesheet or add `<script>` from local js file from that index.html?

**Comment by dev-journey** [comment]
Yes, I think I could do that.

**Comment by valdirunars** [comment]
Oh yeah, my bad, I see this is now fixed. When, I wonder.
